extensions/arc/src/test/stubs.ts

```diff
@@ -39,26 +51,44 @@ export function setupMockComponentBuilder<T extends azdata.Component, P extends
 	mockComponentBuilder?: TypeMoq.IMock<B>,
 ): TypeMoq.IMock<B> {
 	mockComponentBuilder = mockComponentBuilder ?? TypeMoq.Mock.ofType<B>();
-	const returnComponent = TypeMoq.Mock.ofType<T>();
-	// Need to setup 'then' for when a mocked object is resolved otherwise the test will hang : https://github.com/florinn/typemoq/issues/66
-	returnComponent.setup((x: any) => x.then).returns(() => { });
+	setupMockComponentBuilderAndComponent<T, P, B>(mockComponentBuilder, componentGetter);
+	return mockComponentBuilder;
+}
+
+function setupMockComponentBuilderAndComponent<T extends azdata.Component, P extends azdata.ComponentProperties, B extends azdata.ComponentBuilder<T, P> = azdata.ComponentBuilder<T, P>>(
+	mockComponentBuilder?: TypeMoq.IMock<B>,
+	componentGetter?: ((props: P) => T)
+) {
+	mockComponentBuilder = mockComponentBuilder ?? TypeMoq.Mock.ofType<B>();
+	const mockComponent = createComponentMock<T>();
 	let compProps: P;
 	mockComponentBuilder.setup(b => b.withProperties(TypeMoq.It.isAny())).callback((props: P) => compProps = props).returns(() => mockComponentBuilder!.object);
 	mockComponentBuilder.setup(b => b.component()).returns(() => {
-		return componentGetter ? componentGetter(compProps) : Object.assign<T, P>(Object.assign({}, returnComponent.object), compProps);
+		return componentGetter ? componentGetter(compProps) : Object.assign<T, P>(Object.assign({}, mockComponent.object), compProps);
 	});
 
 	// For now just have these be passthrough - can hook up additional functionality later if needed
 	mockComponentBuilder.setup(b => b.withValidation(TypeMoq.It.isAny())).returns(() => mockComponentBuilder!.object);
-	return mockComponentBuilder;
+	return { mockComponentBuilder, mockComponent };
```

Comment:
These should really have field names (builder and component for example)

Reply:
the default implicit field names are the names of the variables, so it would be mockComponentBuilder and mockComponent. I find the deafaults easy and intuitive to use.